pkg/network/ovn_kubernetes.go

```diff
@@ -212,6 +215,26 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	return objs, nil
 }
 
+// returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
+// if it exists, otherwise returns whatever the global OVN_GATEWAY_MODE is set to (shared)
+func GetGatewayMode(kubeClient client.Client) string {
+	cm := &corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
+		ObjectMeta: metav1.ObjectMeta{Name: "gateway-mode-config"},
+	}
+	nsn := types.NamespacedName{Namespace: "openshift-network-operator", Name: "gateway-mode-config"}
+	err := kubeClient.Get(context.TODO(), nsn, cm)
+	if err != nil {
```

Comment:
You have to check to see if this is a not-found error. Only then can you safely return the default. Otherwise you have to return `err` and stop rendering.

Otherwise we could flap gateway modes (bad) or do the wrong thing if the apiserver blips.

Reply:
something like
```
import (
    apierrors "k8s.io/apimachinery/pkg/api/errors"
)

...
if apierrors.IsNotFound(err) {
    gatewayMode = DEFAULT_GATEWAY_MODE
} else {
    return "", fmt.Errorf("could not determine gateway mode: %w", err)
}